EIPS/eip-4788.md

```diff
@@ -29,7 +29,8 @@ restaking constructions, smart contract bridges, MEV mitigations and more.
 |---                           |---                                           |---
 | `FORK_TIMESTAMP`             | TBD                                          |
 | `HISTORY_STORAGE_ADDRESS`    | `Bytes20(0xB)`                               |
-| `G_beacon_root`              | 2100                                         | gas
+| `G_beacon_root`              | 4200                                         | gas
+| `HISTORICAL_ROOTS_LENGTH`    | 8192                                         |
```

Comment:
note: this provides just a bit over 1 day of history, the only downside I can see here is if a user needed to verify something against a root and was facing censorship although I think we generally consider it too expensive to censor for an entire day

Reply:
I think currently this is not correct. If we have this as key for timestamps: `timestamp_reduced = block_header.timestamp % HISTORICAL_ROOTS_LENGTH`, then this would actually generate 2048 unique keys to store this timestamp on, rather than 8192 (which means only ~6.82 hours are saved). I think what is meant here is that 8192 roots are saved, so the calculation of the timestamp key should adapt here to ensure that 8192 keys can fit here.

Reply:
yep good catch, this commit should actually just be rolled back; I was talking to someone about this and convinced myself I had made an error but the original value is correct, thanks!